"AssertionError: the constraints must have the same normal direction"  after  passed "samedir" check.

```julia
HPolytope{Float64,Array{Float64,1}}(HalfSpace{Float64,Array{Float64,1}}[HalfSpace{Float64,Array{Float64,1}}([1.0, 0.0], 1.0), HalfSpace{Float64,Array{Float64,1}}([0.0, 1.0], 1.0), HalfSpace{Float64,Array{Float64,1}}([-1.0, 0.0], 1.0), HalfSpace{Float64,Array{Float64,1}}([0.0, -1.0], 1.0), HalfSpace{Float64,Array{Float64,1}}([0.17178027783046604, -0.342877222169534], -0.3988040749330524), HalfSpace{Float64,Array{Float64,1}}([0.17228094170254737, -0.3438765582974526], -0.5161575)])

AssertionError: the constraints must have the same normal direction

Stacktrace:
 [1] macro expansion at /home/wth/.julia/packages/LazySets/FwmiZ/src/Assertions/Assertions.jl:23 [inlined]
 [2] is_tighter_same_dir_2D(::HalfSpace{Float64,Array{Float64,1}}, ::HalfSpace{Float64,Array{Float64,1}}; strict::Bool) at /home/wth/.julia/packages/LazySets/FwmiZ/src/Sets/HalfSpace.jl:460
 [3] isredundant(::HalfSpace{Float64,Array{Float64,1}}, ::HalfSpace{Float64,Array{Float64,1}}, ::HalfSpace{Float64,Array{Float64,1}}) at /home/wth/.julia/packages/LazySets/FwmiZ/src/Interfaces/AbstractHPolygon.jl:348
 [4] addconstraint!(::Array{HalfSpace{Float64,Array{Float64,1}},1}, ::HalfSpace{Float64,Array{Float64,1}}; linear_search::Bool, prune::Bool) at /home/wth/.julia/packages/LazySets/FwmiZ/src/Interfaces/AbstractHPolygon.jl:521
 [5] #addconstraint!#72 at /home/wth/.julia/packages/LazySets/FwmiZ/src/Interfaces/AbstractHPolygon.jl:434 [inlined]
 [6] convert(::Type{HPolygon}, ::HPolytope{Float64,Array{Float64,1}}; prune::Bool) at /home/wth/.julia/packages/LazySets/FwmiZ/src/convert.jl:239
 [7] vertices_list(::HPolytope{Float64,Array{Float64,1}}; backend::Nothing, prune::Bool) at /home/wth/.julia/packages/LazySets/FwmiZ/src/Sets/HPolytope.jl:218
 [8] vertices_list at /home/wth/.julia/packages/LazySets/FwmiZ/src/Sets/HPolytope.jl:213 [inlined]
 [9] area(::HPolytope{Float64,Array{Float64,1}}) at /home/wth/.julia/packages/LazySets/FwmiZ/src/Interfaces/LazySet.jl:944
 [10] ordinal_split!(::IntervalNet, ::Problem{HPolytope{Float64,Array{Float64,1}},HPolytope{Float64,Array{Float64,1}}}, ::NeuralVerification.Tree{Tuple{HPolytope{Float64,Array{Float64,1}},Array{Any,1}}}, ::Int64, ::Int64, ::Array{Tuple{Int64,Int64},1}) at /home/wth/dev/NeuralVerification.jl/src/runtime/trainingVerification.jl:59
 [11] init_split at /home/wth/dev/NeuralVerification.jl/src/runtime/trainingVerification.jl:91 [inlined]
 [12] solve(::TrainingProblem{HPolytope{Float64,Array{Float64,1}},HPolytope{Float64,Array{Float64,1}}}, ::Int64, ::Bool, ::Bool, ::Float64, ::Bool) at /home/wth/dev/NeuralVerification.jl/src/runtime/trainingVerification.jl:320
 [13] macro expansion at ./timing.jl:310 [inlined]
 [14] test2D(::String, ::Int64, ::Bool, ::Bool, ::Float64, ::Bool) at ./In[3]:42
 [15] test2D(::String, ::Int64, ::Bool, ::Bool, ::Float64) at ./In[3]:28
 [16] top-level scope at In[20]:1
 [17] include_string(::Function, ::Module, ::String, ::String) at ./loading.jl:1091
```

--- a/src/Arrays/Arrays.jl
+++ b/src/Arrays/Arrays.jl
@@ -9,7 +9,7 @@ using LinearAlgebra, SparseArrays
 using ..Assertions: @assert, activate_assertions
 # activate assertions by default
 activate_assertions(Arrays)
-using ..LazySets: _geq, isapproxzero
+using ..LazySets: _geq, isapproxzero, _isapprox
 
 include("matrix_operations.jl")
 include("vector_operations.jl")

--- a/src/Arrays/vector_operations.jl
+++ b/src/Arrays/vector_operations.jl
@@ -111,7 +111,7 @@ function samedir(u::AbstractVector{N},
             if factor < 0
                 return (false, 0)
             end
-        elseif factor != u[i] / v[i]
+        elseif !_isapprox(factor, u[i] / v[i])
             return (false, 0)
         end
     end

--- a/test/unit_Polygon.jl
+++ b/test/unit_Polygon.jl
@@ -540,6 +540,16 @@ for N in [Float64]
        ]
     P = HPolygon(copy(Hs))
     @test ispermutation(P.constraints, Hs[3:6])
+
+    # redundancy with almost-parallel constraints and approximation issues (#2386)
+    P = HPolygon([HalfSpace([1.0, 0.0], 1.0),
+                  HalfSpace([0.0, 1.0], 1.0),
+                  HalfSpace([-1.0, 0.0], 1.0),
+                  HalfSpace([0.0, -1.0], 1.0),
+                  HalfSpace([0.17178027783046604, -0.342877222169534], -0.3988040749330524),
+                  HalfSpace([0.17228094170254737, -0.3438765582974526], -0.5161575)
+                 ])
+    @test vertices_list(P) == [[-1.0, 1.0]]
 end
 
 # default Float64 constructors

